### PR TITLE
[mds-compliance] Fix compliance maximum 0 evaluation, and add test case to explicitly exercise that behavior. Thanks javascript <3

### PR DESCRIPTION
**Pull request by avatarneil**
PS: I hate javascript :) 
## PR Checklist

 - [x] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [x] briefly describe the changes in this PR
 - [x] write tests for all new functionality

## Impacts
- [x] Compliance

### CODE REVIEW

File: packages/mds-compliance/tests/api.spec.ts [hunk 1]
```diff
@@ -1088,4 +1113,33 @@ describe('Tests Compliance API:', () => {
         })
     })
   })
+
+  describe('Verifies max 0 count policy', () => {
+    before('boop', async () => {
```
Comment (janedotx) [inline packages/mds-compliance/tests/api.spec.ts]: you cannot name it boop

Comment (avatarneil) [reply packages/mds-compliance/tests/api.spec.ts]: I mean, I _CAN_. Though I probably shouldn't 😜

File: packages/mds-compliance/mds-compliance-engine.ts [hunk 2]
```diff
@@ -245,7 +245,8 @@ function processPolicy(
                     match_instance: MatchedVehicle,
                     i: number
                   ) => {
-                    const maximum = rule.maximum || Number.POSITIVE_INFINITY
+                    // If the rule has a defined maximum, use it, even if 0
+                    const maximum = rule.maximum == null ? Number.POSITIVE_INFINITY : rule.maximum
```
Comment (wesleyfsmith) [inline packages/mds-compliance/mds-compliance-engine.ts]: So if they don't define a max the rule is to let them have an unlimited number of scooters in the area? I would that we would want to require them to set some sort of upper bound, but idk

Comment (avatarneil) [reply packages/mds-compliance/mds-compliance-engine.ts]: This is primarily so we have a way to encode endless size buckets, for the case in which we might want to catch all vehicles and prevent further rule evaluation for them. Unfortunately there's no way to specify a POS_INF in JSON, so we take a count rule with a lack of a maximum as there being an endlessly sized bucket that vehicles can fall into during evaluation.

Comment (avatarneil) [reply packages/mds-compliance/mds-compliance-engine.ts]: So, say for example I have two geographies A and B, where A is a subset of B (so like a bunch of valid drop off zones with no cap inside of a park which has greater regulations, for example). You could encode a policy that allows them to have as many vehicles as they want in the valid drop off zones `geography: A, max: null/undef` for the first rule, and then `geography: B, max: 0` for the second.

**Pull request merged**